### PR TITLE
Fix flake8 errors on sklearn integration

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

After merged #4379, checks action become being failed.
https://github.com/optuna/optuna/actions/runs/4130385467/jobs/7137016344

## Description of the changes

Add an import statement of sklearn.

### CODE REVIEW

File: .github/workflows/checks-integration.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ on:
   push:
     branches:
       - master
+  # TODO(c-bata): Revert this temporary change after checking this PR.
+  pull_request: {}
```
Comment (c-bata) [inline .github/workflows/checks-integration.yml:8]:
```suggestion
```

Comment (c-bata) [reply .github/workflows/checks-integration.yml:8]: It seems to have been resolved.

```
Run mypy . --warn-unused-ignores
optuna/integration/pytorch_distributed.py:137: error: Incompatible types in assignment (expression has type "Optional[ProcessGroup]", variable has type "ProcessGroup")  [assignment]
optuna/integration/pytorch_lightning.py:98: error: "Trainer" has no attribute "training_type_plugin"  [attr-defined]
optuna/integration/allennlp/_pruner.py:26: error: Module "allennlp.training" does not explicitly export attribute "GradientDescentTrainer"  [attr-defined]
optuna/integration/allennlp/_pruner.py:27: error: Module "allennlp.training" does not explicitly export attribute "TrainerCallback"  [attr-defined]
tests/integration_tests/test_chainer.py:23: error: Cannot assign to a type  [misc]
tests/integration_tests/test_chainer.py:23: error: Incompatible types in assignment (expression has type "Type[object]", variable has type "Type[DatasetMixin]")  [assignment]
tests/integration_tests/allennlp_tests/test_allennlp.py:303: error: "Module allennlp.training" does not explicitly export attribute "GradientDescentTrainer"  [attr-defined]
Found 7 errors in 5 files (checked 302 source files)
Error: Process completed with exit code 1.
```

https://github.com/optuna/optuna/actions/runs/4130496890/jobs/7137247993

**Review by toshihikoyanase — APPROVED** [review]
`import sklearn` is required to find `sklearn.base.BaseEstimator`.
I confirmed that this change fixes the following `flake8` errors:

```console
% flake8 optuna/integration/sklearn.py                                                                                                                 
optuna/integration/sklearn.py:182:20: F821 undefined name 'sklearn'
optuna/integration/sklearn.py:234:40: F821 undefined name 'sklearn'
optuna/integration/sklearn.py:287:20: F821 undefined name 'sklearn'
optuna/integration/sklearn.py:675:20: F821 undefined name 'sklearn'
```

And `sklearn_optuna_search_cv_simple.py` in the optuna examples worked as expected.
LGTM!

**Comment by toshihikoyanase** [comment]
@contramundum53 Could you review this PR, please?

**Review by contramundum53 — APPROVED** [review]
LGTM.

**Pull request merged**